src/window-dimensions.ts

```diff
@@ -1,19 +1,32 @@
-import { useEffect, useState } from 'react';
+import { useEffect, useMemo, useState } from 'react';
 import { Dimensions } from 'react-native';
 
 import type { ScaledSize } from 'react-native';
+import { ComponentSizeChanges } from 'make-styles';
 
 const WINDOW_DIMENSIONS_DEBOUNCE = 500;
 const windowDimensionListeners = new Map();
 let initialWindowDimensions = Dimensions.get('window');
-let timeoutId: NodeJS.Timeout | undefined;
-Dimensions.addEventListener('change', ({ window }) => {
-  timeoutId && clearTimeout(timeoutId);
-  timeoutId = setTimeout(() => {
-    initialWindowDimensions = window;
-    windowDimensionListeners.forEach((listener) => listener(window));
-  }, WINDOW_DIMENSIONS_DEBOUNCE);
-});
+
+const debounce = (fn: (...args: any[]) => void, duration = 100) => {
```

Comment:
this might belong in a separate util and the `useContainerDimensions` doesn't have much to do with window dimensions, I wouldn't put all this code in one file probably because it has no semantic connection